go/vt/vtgate/planbuilder/ddl.go

```diff
@@ -25,6 +25,9 @@ const (
 // and which chooses which of the two to invoke at runtime.
 func buildGeneralDDLPlan(sql string, ddlStatement sqlparser.DDLStatement, vschema ContextVSchema) (engine.Primitive, error) {
 	normalDDLPlan, onlineDDLPlan, err := buildDDLPlans(sql, ddlStatement, vschema)
+	if vschema.Destination() != nil {
+		return buildByPassDDLPlan(sql, vschema)
+	}
```

Comment:
this check should move before calling `buildDDLPlans`

Reply:
oops - I resolved the conflict too blindly. Fixing